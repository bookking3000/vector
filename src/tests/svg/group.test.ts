import { Group, SVG } from '../..';

let group : Group;

beforeEach(() => {
  let container = document.createElement('div');
  document.body.appendChild(container);
  let svg = SVG.SVG(container);
  group = svg.group();
});

describe('group', () => {

  it('should render a circle with cx=1 cy=2 and r=3', () => {
    const element = group.circle(1,2,3);
    expect(document.getElementById(element.id)).toMatchSnapshot();
  });

  it('should render an ellipse with cx=1 cy=2 rx=3 and ry=4', () => {
    const element = group.ellipse(1,2,3,4);
    expect(document.getElementById(element.id)).toMatchSnapshot();
  });

  it('should render a line with x1=1 y1=2 x2=3 and y2=4', () => {
    const element = group.line(1,2,3,4);
    expect(document.getElementById(element.id)).toMatchSnapshot();
  });

  it('should render a path with d="M1 2 L 2 3"', () => {
    const element = group.path('M1 2 L 3 4');
    expect(document.getElementById(element.id)).toMatchSnapshot();
  });

  it('should render a polygon with points="1,2 3,4"', () => {
    const element = group.polygon('1,2 3,4');
    expect(document.getElementById(element.id)).toMatchSnapshot();
  });

  it('should render a rectangle with x=1 y=2 width=3 and height=4', () => {
    const element = group.rectangle(1,2,3,4);
    expect(document.getElementById(element.id)).toMatchSnapshot();
  });
  
});