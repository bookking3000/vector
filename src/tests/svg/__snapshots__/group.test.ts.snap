// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`group should render a circle with cx=1 cy=2 and r=3 1`] = `
<circle
  cx="1"
  cy="2"
  id="circle-2"
  r="3"
/>
`;

exports[`group should render a line with x1=1 y1=2 x2=3 and y2=4 1`] = `
<line
  id="line-8"
  x1="1"
  x2="3"
  y1="2"
  y2="4"
/>
`;

exports[`group should render a path with d="M1 2 L 2 3" 1`] = `
<path
  d="M1 2 L 3 4"
  id="path-11"
/>
`;

exports[`group should render a polygon with points="1,2 3,4" 1`] = `
<polygon
  id="polygon-14"
  points="1,2 3,4"
/>
`;

exports[`group should render a rectangle with x=1 y=2 width=3 and height=4 1`] = `
<rect
  height="4"
  id="rectangle-17"
  width="3"
  x="1"
  y="2"
/>
`;

exports[`group should render an ellipse with cx=1 cy=2 rx=3 and ry=4 1`] = `
<ellipse
  cx="1"
  cy="2"
  id="ellipse-5"
  rx="3"
  ry="4"
/>
`;
